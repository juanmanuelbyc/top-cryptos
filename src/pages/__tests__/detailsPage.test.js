import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import store from '../../redux/configStore';
import DetailsPage from '../detailsPage';
import '@testing-library/jest-dom/extend-expect';

const TestDetailsPage = () => {
  const history = createMemoryHistory();
  return (
    <Provider store={store}>
      <Router location={history.location} navigator={history}>
        <DetailsPage />
      </Router>
    </Provider>
  );
};

render(<TestDetailsPage />);

describe('Details page', () => {
  it('gets elements', () => {
    expect(screen.getByText('current_price')).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <TestDetailsPage />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
