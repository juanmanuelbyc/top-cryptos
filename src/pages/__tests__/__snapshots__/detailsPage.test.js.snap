// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page renders correctly 1`] = `
<div
  className="detailsContainer"
>
  <div
    className="facade"
  >
    <div
      className="facadeTopSection"
    >
      <a
        className="backHomeButton"
        href="/Home"
        onClick={[Function]}
      >
        ‚Üê
      </a>
      <p
        className="facadeTopText"
      >
        cryptocurrency details
      </p>
      <div
        className="topFacadeIcons"
      >
        <img
          alt="mic Icon"
          className="micIcon"
          src="micIcon.png"
        />
        <img
          alt="settings Icon"
          className="settingsIcon"
          src="settingsIcon.png"
        />
      </div>
    </div>
    <div
      className="facadeMiddleSection"
    >
      <div
        className="facadeImageContainer"
      >
        <img
          alt="cryptoImage"
          className="facadeImage"
          src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
        />
      </div>
      <div
        className="facadeTextsContainer"
      >
        <h2
          className="facadeMainText"
        >
          BITCOIN
        </h2>
      </div>
    </div>
    <div
      className="facadeBottomSection"
    >
      <p>
        UPDATED STATISTICS
      </p>
    </div>
  </div>
  <div
    className="detailCard"
  >
    <h3
      className="detailCardText"
    >
      current_price
    </h3>
    <h3
      className="detailCardText"
    >
      20369
    </h3>
  </div>
</div>
`;
